From 335ad1a53f5d28f7e8fc09db09accc56313ac90e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 9 Oct 2025 23:36:26 +0200
Subject: [PATCH] Set XDG_CURRENT_DESKTOP for new units started by user service
 manager

This fixes the problem that XDG Desktop Portal does not use the LXQt backend under LXQt session.
---
 startlxqtwayland.in | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/startlxqtwayland.in b/startlxqtwayland.in
index 64d2b3a..2829685 100644
--- a/startlxqtwayland.in
+++ b/startlxqtwayland.in
@@ -83,6 +83,11 @@ done
 
 export XDG_CURRENT_DESKTOP="LXQt:$COMPOSITOR:wlroots"
 
+# set environment variables for new units started by user service manager
+if command -v systemctl >/dev/null; then
+    systemctl --user import-environment XDG_CURRENT_DESKTOP
+fi
+
 valid_layouts=$(grep -A98 '! layout' /usr/share/X11/xkb/rules/base.lst | awk '{print $1}' | grep -v '!')
 trylayout=$(echo $LANG | cut -c 1,2)
 
@@ -91,6 +96,11 @@ if  [ -z "$COMPOSITOR" ]; then
     COMPOSITOR=labwc
     export XDG_CURRENT_DESKTOP="LXQt:wlroots"
 
+    # set environment variables for new units started by user service manager
+    if command -v systemctl >/dev/null; then
+        systemctl --user import-environment XDG_CURRENT_DESKTOP
+    fi
+
     # enable cursor on VM (systemd only)
     if type systemd-detect-virt > /dev/null 2>&1 && systemd-detect-virt --quiet; then
         export WLR_NO_HARDWARE_CURSORS=1
@@ -125,6 +135,12 @@ elif [ "$COMPOSITOR" = "kwin_wayland" ]; then
     # Style KDE's QML apps like systemsettings
     export QT_QUICK_CONTROLS_STYLE=org.kde.desktop
     export XDG_CURRENT_DESKTOP="LXQt:$COMPOSITOR"
+
+    # set environment variables for new units started by user service manager
+    if command -v systemctl >/dev/null; then
+        systemctl --user import-environment XDG_CURRENT_DESKTOP
+    fi
+
     if echo "$valid_layouts" | grep -q "$trylayout"; then
         kxkbrc=$XDG_CONFIG_HOME/kxkbrc
         layout="LayoutList=$trylayout"
